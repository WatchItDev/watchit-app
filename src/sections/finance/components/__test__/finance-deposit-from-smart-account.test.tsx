import { render } from "@testing-library/react";
import { describe, expect, it, vi } from "vitest";
import FinanceDepositFromSmartAccount from "../finance-deposit-from-smart-account";

vi.mock("@src/hooks/use-auth.ts", () => ({
  useAuth: () => ({
    session: {
      address: "0x123456789abcdef",
    },
  }),
}));

vi.mock("@src/hooks/protocol/use-deposit.ts", () => ({
  useDeposit: () => ({
    mutate: vi.fn(),
    isLoading: false,
  }),
}));

vi.mock("../finance-deposit", () => ({
  default: () => <div data-testid="FinanceDeposit">Mocked FinanceDeposit</div>,
}));

const mockOnClose = vi.fn();
const renderComponent = () => render(<FinanceDepositFromSmartAccount onClose={mockOnClose} />);

describe("FinanceDepositFromSmartAccount", () => {
  it("to match snapshot", () => {
    const { container } = renderComponent();
    expect(container).toMatchSnapshot();
  });
});
