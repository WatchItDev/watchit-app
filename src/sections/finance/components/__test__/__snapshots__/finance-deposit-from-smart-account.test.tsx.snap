// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`FinanceDepositFromSmartAccount > should render correctly and match snapshot 1`] = `
<div>
  <div
    data-testid="FinanceDeposit"
  >
    Mocked FinanceDeposit
  </div>
</div>
`;
